hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsManagedApacheHttpConnection.java

```diff
@@ -91,7 +97,27 @@ public boolean isOpen() {
   /**{@inheritDoc}*/
   @Override
   public boolean isStale() {
-    return httpClientConnection.isStale();
+    try {
```

Comment:
any specific reason for not using the isStale existing method and implementing a custom approach ?